src/main/java/uk/ac/cam/cl/dtg/segue/dao/users/PgTOTPDataManager.java

```diff
@@ -0,0 +1,237 @@
+        } catch (SQLException e) {
+            throw new SegueDatabaseException("Postgres exception", e);
+        }
+    }
+
+    /**
+     * Remove all trace of MFA for the user account.
+     *
+     * @param userId of the user to affect.
+     * @throws SegueDatabaseException - if we have a problem with the database connection.
+     */
+    @Override
+    public void delete2FACredentials(final Long userId) throws SegueDatabaseException {
+
+        PreparedStatement pst;
+        try (Connection conn = database.getDatabaseConnection()) {
+            pst = conn
+                    .prepareStatement(
+                            "DELETE FROM user_totp "
+                                    + "WHERE user_id = ?;");
+
+            setValueHelper(pst, 1, userId);
+
+            pst.executeUpdate();
+
+        } catch (SQLException e) {
+            throw new SegueDatabaseException("Postgres exception", e);
+        }
+    }
+
+    /**
+     * findOne helper method to ensure that only one result matches the search criteria.
+     *
+     * @param results
+     *            - from a jdbc database search
+     * @return a single TOTPSharedSecret that matches the search criteria or null of no matches found.
+     * @throws SQLException
+     *             - if a db error occurs
+     * @throws SegueDatabaseException
+     *             - if more than one result is returned
+     */
+    private TOTPSharedSecret findOne(final ResultSet results) throws SegueDatabaseException, SQLException {
+        // are there any results
+        if (!results.isBeforeFirst()) {
+            return null;
+        }
+
+        List<TOTPSharedSecret> listOfResults = Lists.newArrayList();
+        while (results.next()) {
+            listOfResults.add(buildTOTPSharedSecret(results));
+        }
+
+        if (listOfResults.size() > 1) {
+            throw new SegueDatabaseException("Ambiguous result, expected single result and found more than one"
+                    + listOfResults);
+        }
+
+        return listOfResults.get(0);
+    }
+
+    /**
+     * Construct an appropriate POJO.
+     *
+     * @param results - sql results
+     * @return totpSharedSecret object
+     * @throws SQLException if we can't get a value required.
+     */
+    private TOTPSharedSecret buildTOTPSharedSecret(final ResultSet results) throws SQLException {
+
+        return new TOTPSharedSecret(results.getLong("user_id"),
+                results.getString("shared_secret"),
+                results.getTimestamp("created"));
+    }
+
+    /**
+     * Helper that picks the correct pst method based on the value provided.
+     *
+     * @param pst - prepared statement - already initialised
+     * @param index - index of the value to be replaced in the pst
+     * @param value - value
+     * @throws SQLException
+     */
+    private void setValueHelper(final PreparedStatement pst, final int index, final Object value) throws SQLException {
```

Comment:
Don't we have at least two of these helper functions in other files already? Can we not add a static method to a class in the `utils` package?